dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java

```diff
@@ -54,14 +63,36 @@ public void serialize(final PageView pageView, final JsonGenerator jsonGenerator
 
         if (null != pageView.getUrlContent()) {
 
-            final DotContentletTransformer transformer = new DotTransformerBuilder().defaultOptions().content(pageView.getUrlContent()).build();
-            final Map<String, Object>   urlContentletMap = transformer.toMaps().stream().findFirst().orElse(Collections.EMPTY_MAP);
-            pageViewMap.put("urlContentMap", urlContentletMap);
+            this.createObjectMapUrlContent(pageView.getUrlContent(), pageViewMap);
         }
 
         return pageViewMap;
     }
 
+    protected void createObjectMapUrlContent(final Contentlet urlContent, final Map<String, Object> pageViewMap) {
+
+        final DotContentletTransformer transformer   = new DotTransformerBuilder().defaultOptions().content(urlContent).build();
+        final Map<String, Object>   urlContentletMap = transformer.toMaps().stream().findFirst().orElse(Collections.EMPTY_MAP);
+        this.createObjectMapKeyValue(urlContent, urlContentletMap);
+
+        pageViewMap.put("urlContentMap", urlContentletMap);
+    }
+
+    /**
+     * Get all key value fields into the contentlet (if any) and put as an object into the contentMap (returns it)
+     * @param contentlet    {@link Contentlet}
+     * @param contentType   {@link ContentType}
+     * @param contentletMap {@link Map}
+     * @return Map
+     */
+    protected Map<String, Object> createObjectMapKeyValue(final Contentlet contentlet,
+                                                          final Map<String, Object> contentletMap) {
+
+        new KeyValueViewStrategy(null).apply(contentlet, contentletMap, null, null);
+        return contentletMap;
+    }
```

Comment:
I don't think the strategy should be applied directly it must be applied from within the transformer. 

Let's say you do
`new DotTransformerBuilder().defaultOptions() `
the code that instructs the transformer to apply this strategy must be part of the options provided to the transformer. You should never apply a strategy outside of the transformer